engine/src/test/scala/cromwell/engine/io/IoActorSpec.scala

```diff
@@ -221,10 +222,26 @@ class IoActorSpec extends TestKitSuite with FlatSpecLike with Matchers with Impl
       new StorageException(429, "message"),
       BatchFailedException(new Exception),
       new SocketException(),
-      new SocketTimeoutException()
+      new SocketTimeoutException(),
+      new IOException("text Error getting access token for service account some other text"),
+      new IOException("Could not read from gs://fc-secure-<snip>/JointGenotyping/<snip>/call-HardFilterAndMakeSitesOnlyVcf/shard-4688/rc: 500 Internal Server Error Backend Error"),
+      new IOException("message: 500 Internal Server Error Backend Error"),
+      new Exception("500 Internal Server Error"),
+      new Exception("501 Not Implemented"),
+      new Exception("510 Extensions are Missing")
     )
 
     retryables foreach { IoActor.isRetryable(_) shouldBe true }
     retryables foreach { IoActor.isFatal(_) shouldBe false }
   }
+
+  it should "have correct non-retryable exceptions" in {
+    val nonRetryables = List(
+      new IOException("404 File Not Found"),
+      new Exception("5xx HTTP Status Code")
```

Comment:
I think your regex will correctly not identify this, but can we make sure with one more test case?

`new IOException("Could not read from gs://fc-secure-<snip>/JointGenotyping/<snip>/call-HardFilterAndMakeSitesOnlyVcf/shard-500/rc: 404 File Not Found")`

Reply:
Good point. I added a new test case for it.